tools/update-authors.js

```diff
@@ -22,6 +23,41 @@ else
 
 output.write('# Authors ordered by first contribution.\n\n');
 
+const mailmap = new Map();
+{
+  const lines = fs.readFileSync(path.resolve(__dirname, '../', '.mailmap'),
+                                { encoding: 'utf8' }).split('\n');
+  for (let line of lines) {
+    line = line.trim();
+    if (line.startsWith('#') || line === '') continue;
+    // Replaced Name <original@example.com>
+    const emailToAuthor = line.match(/^([^<]+)\s+(<[^>]+>)$/);
+    // <replaced@example.com> <original@example.com>
+    const emailToEmail = line.match(/^<([^>]+)>\s+(<[^>]+>)$/);
+    // Replaced Name <replaced@example.com> <original@example.com>
+    const emailToBoth = line.match(/^([^<]+)\s+(<[^>]+>)\s+(<[^>]+>)$/);
+    // Replaced Name <replaced@example.com> Original Name <original@example.com>
+    const bothToBoth =
+        line.match(/^([^<]+)\s+(<[^>]+>)\s+([^<]+)\s+(<[^>]+>)$/);
```

Comment:
This is not really performance critical but wouldn't it be better to just inline the regular expressions with a comment next to it what it does? That way we do not execute each of them on all lines if e.g., the first one already matched.